src/SignalR/server/Core/src/HubClientsBase`T.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    /// <summary>
+    /// An abstraction that provides access to client connections.
+    /// </summary>
+    /// <typeparam name="T">The client invoker type.</typeparam>
+    public abstract class HubClientsBase<T>
```

Comment:
What do we think about removing the Base suffix from all these abstract classes? So this would just be `HubClients<T>`, `HubCallerClientsBase<T>` would be `HubCallerClients<T>` etc...

Reply:
Also, even though it's obsolete, this should implement `IHubClients<T>`. All the new abstract base classes should implement the interfaces they're replacing for max compatibility until the interfaces are fully removed.

Reply:
> So this would just be HubClients<T>, HubCallerClientsBase<T> ...

We already have the concrete class `HubCallerClients`. That's why I added the base suffix.